cmd/osbuild-auth-tests/certificates.go

```diff
@@ -28,61 +34,153 @@ func (ckp certificateKeyPair) key() string {
 	return path.Join(ckp.baseDir, "key")
 }
 
-func newCertificateKeyPair(CA, CAkey, subj string) (*certificateKeyPair, error) {
+func newSelfSignedCertificateKeyPair(subj string) (*certificateKeyPair, error) {
 	dir, err := ioutil.TempDir("", "osbuild-auth-tests-")
 	if err != nil {
 		return nil, fmt.Errorf("cannot create a temporary directory for the certificate: %v", err)
 	}
 
 	ckp := certificateKeyPair{baseDir: dir}
-	certificateRequest := path.Join(dir, "csr")
 
 	cmd := exec.Command(
-		"openssl", "req", "-new", "-nodes",
+		"openssl", "req", "-nodes", "-x509",
 		"-subj", subj,
+		"-out", ckp.certificate(),
 		"-keyout", ckp.key(),
-		"-out", certificateRequest,
 	)
-
 	err = cmd.Run()
 	if err != nil {
-		return nil, fmt.Errorf("cannot generate a private key and a certificate request: %v", err)
+		return nil, fmt.Errorf("cannot generate a self-signed certificate: %v", err)
 	}
 
-	defer os.Remove(certificateRequest)
+	return &ckp, nil
+}
 
-	cmd = exec.Command(
-		"openssl", "x509", "-req", "-CAcreateserial",
-		"-in", certificateRequest,
-		"-CA", CA,
-		"-CAkey", CAkey,
-		"-out", ckp.certificate(),
+type ca struct {
+	BaseDir string
+}
+
+func (c ca) remove() {
+	err := os.RemoveAll(c.BaseDir)
+	if err != nil {
+		log.Printf("cannot delete the ca: %v", err)
+	}
+}
+
+func (c ca) certificate() string {
+	return path.Join(c.BaseDir, "ca.cert.pem")
+}
+
+func (c ca) key() string {
+	return path.Join(c.BaseDir, "private", "ca.key.pem")
+}
+
+func newCA(subj string) (*ca, error) {
```

Comment:
Could we just generate all these things in the provision script and not anything from golang?

Reply:
We definitely can, I'm just not sure how to do this elegantly in `bash` without repeating many lines of `openssl` calls. Any ideas?

Reply:
Could have a little helper for this? Either a function or a separate script. Not sure what's best to be honest.

Reply:
I think I'd prefer having this in the provision script as well, but wouldn't block on it. Not a regression.